include/csoundCore.h

```diff
@@ -203,11 +203,11 @@ extern "C" {
 #define CURTIME (((double)csound->icurTime)/((double)csound->esr))
 #define CURTIME_inc (((double)csound->ksmps)/((double)csound->esr))
 
-#ifdef  B64BIT
-#define MAXLEN     0x10000000
-#define FMAXLEN    ((MYFLT)(MAXLEN))
-#define PHMASK     0x0fffffff
-#else
+#ifndef  SHORT_TABLE_LENGTH  // long max table length is the default
+static const uint32_t MAXLEN = 1U << 31;  
```

Comment:
General question why we're using uint32_t over uint64_t. Kind of odd we're mixing 32-bit ints with MYFLT and double operations.

Reply:
Ok so the table length is used in the fixed point computing and that's strictly a 32-bit size (that's why the 2^31 limit). If we change this here, then we will need to make changes throughout the codebase (to avoid warnings at least I think but more so to be able to use 64 bis so push MAXLEN higher). For now, I think this is a good compromise.

FMAXLEN needs to accommodate an exponent size that is potentially bigger than in float, so I thought it should be double in order not to loose precision. This is also the case in the original code, but I am not proposing changing that.